### PR TITLE
[boost] update to 1.80.0

### LINKED ISSUE #26282: [boost] update to 1.80
Library name: Boost

New version number: 1.80

Other information that may be useful (release notes, etc...)

https://www.boost.org/users/history/version_1_80_0.html

### PR DESCRIPTION
**Pull request by yurybura**
This PR updates Boost to v1.80.0.

Fixes #26282

### CODE REVIEW

**Comment by Neumann-A** [comment]
Switch to the CMake build? As far as I could tell 1.80 finally comes with proper CMake scripts which install `-config.cmake` etc.

**Comment by Neumann-A** [comment]
> Could you provide details? [Release notes](https://www.boost.org/users/history/version_1_80_0.html) doesn't contain this information yet.

hmm ok maybe I was just looking at one of the boost modules and it looked like proper CMake. Looked at program_options and it didn't have what I saw the last time so it might still be partial. So just ignore my ramblings.

**Comment by yurybura** [comment]
> ah please consider including/fixing #26004/[#22762 (comment)](https://github.com/microsoft/vcpkg/discussions/22762#discussioncomment-2126877)

I think it is better to fix this issue in a separate MR. Did you report the issue in Boost.Log?

**Comment by Neumann-A** [comment]
That is https://github.com/boostorg/log/issues/177 but it is really a bug in the vcpkg boost setup not carrying required dependencies between boost ports.

File: ports/boost-build/portfile.cmake [hunk 1]
```diff
@@ -4,18 +4,16 @@ if(CMAKE_HOST_WIN32 AND VCPKG_CMAKE_SYSTEM_NAME AND NOT VCPKG_CMAKE_SYSTEM_NAME
     return()
 endif()
 
-set(BOOST_VERSION 1.79.0)
+set(BOOST_VERSION 1.80.0.beta1)
```
Comment (Cheney-W) [inline ports/boost-build/portfile.cmake:7]: Why is it set to `1.80.0.beta1` here, isn't `1.80.0` an official release ?

File: ports/boost-modular-build-helper/boost-modular-build.cmake [hunk 2]
```diff
@@ -5,7 +5,7 @@ include("${CMAKE_CURRENT_LIST_DIR}/../vcpkg-cmake-get-vars/vcpkg-port-config.cma
 get_filename_component(BOOST_BUILD_INSTALLED_DIR "${CMAKE_CURRENT_LIST_DIR}" DIRECTORY)
 get_filename_component(BOOST_BUILD_INSTALLED_DIR "${BOOST_BUILD_INSTALLED_DIR}" DIRECTORY)
 
-set(BOOST_VERSION 1.79.0)
+set(BOOST_VERSION 1.80.0.beta1)
```
Comment (Cheney-W) [inline ports/boost-modular-build-helper/boost-modular-build.cmake:8]: Same question as above.

File: ports/boost-vcpkg-helpers/portfile.cmake [hunk 3]
```diff
@@ -1,6 +1,6 @@
 set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
 
-set(BOOST_VERSION 1.79.0)
+set(BOOST_VERSION 1.80.0.beta1)
```
Comment (Cheney-W) [inline ports/boost-vcpkg-helpers/portfile.cmake:3]: The content of `1.80.0` license_1_0.txt is the same as that of `1.80.0.beta1`, so `1.80.0` should be filled in here.

File: ports/boost-build/portfile.cmake (hunk 1 above)
Comment (yurybura) [reply ports/boost-build/portfile.cmake]: Fixed.

File: ports/boost-modular-build-helper/boost-modular-build.cmake (hunk 2 above)
Comment (yurybura) [reply ports/boost-modular-build-helper/boost-modular-build.cmake]: Fixed.

File: ports/boost-vcpkg-helpers/portfile.cmake (hunk 3 above)
Comment (yurybura) [reply ports/boost-vcpkg-helpers/portfile.cmake]: Fixed.

**Review by BillyONeal — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**